url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/apache/harmony/beans/BeansUtils.java

```diff
@@ -0,0 +1,122 @@
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+
+package org.apache.harmony.beans;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+public class BeansUtils {
+
+    public static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+
+    public static final String NEW = "new"; //$NON-NLS-1$
+
+    public static final String NEWINSTANCE = "newInstance"; //$NON-NLS-1$
+
+    public static final String NEWARRAY = "newArray"; //$NON-NLS-1$
+
+    public static final String FORNAME = "forName"; //$NON-NLS-1$
+
+    public static final String GET = "get"; //$NON-NLS-1$
+
+    public static final String IS = "is"; //$NON-NLS-1$
+
+    public static final String SET = "set"; //$NON-NLS-1$
+
+    public static final String ADD = "add"; //$NON-NLS-1$
+
+    public static final String PUT = "put"; //$NON-NLS-1$
+
+    public static final String NULL = "null"; //$NON-NLS-1$
+
+    public static final String QUOTE = "\"\""; //$NON-NLS-1$
+
+    public static final int getHashCode(Object obj) {
+        return obj != null ? obj.hashCode() : 0;
+    }
+
+    public static final int getHashCode(boolean bool) {
+        return bool ? 1 : 0;
+    }
+
+    public static String toASCIILowerCase(String string) {
+        char[] charArray = string.toCharArray();
+        StringBuilder sb = new StringBuilder(charArray.length);
+        for (int index = 0; index < charArray.length; index++) {
+            if ('A' <= charArray[index] && charArray[index] <= 'Z') {
+                sb.append((char) (charArray[index] + ('a' - 'A')));
+            } else {
+                sb.append(charArray[index]);
+            }
+        }
+        return sb.toString();
+    }
+
+    public static String toASCIIUpperCase(String string) {
+        char[] charArray = string.toCharArray();
+        StringBuilder sb = new StringBuilder(charArray.length);
+        for (int index = 0; index < charArray.length; index++) {
+            if ('a' <= charArray[index] && charArray[index] <= 'z') {
+                sb.append((char) (charArray[index] - ('a' - 'A')));
+            } else {
+                sb.append(charArray[index]);
+            }
+        }
+        return sb.toString();
+    }
+
+    public static boolean isPrimitiveWrapper(Class<?> wrapper, Class<?> base) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 84-91 into a separate method.